### ISSUE TITLE
Unable to get correct typing for protobuf defined in gem

### ISSUE THREAD

**Issue by SamMorrowDrums** [Shopify/tapioca#1780]
May be related to #564 

We have a gem that provides a twirp interface, and when I attempt to view type info via the sorbet plugin, I see only untyped:

The RPC is:

```
  rpc GetAlert(AlertRequest) returns (AlertResponse);
```

![Image](https://github.com/Shopify/tapioca/assets/4811358/f21996da-6ab2-4670-b273-cb19af1e9670)

There is a gem RBI, and a DSL that get generated when we vendor the gem (and there is a conflict, so the gem RBI gets switched to `# typed: false`).

And the function has the following definition in that file:

```
  # source://twirp/1.10.0/lib/twirp/client.rb#42
  def get_alert(input, req_opts = T.unsafe(nil)); end
```

`get_alert` produces a `Turboscan::Proto::AlertRequest`, and then returns a `Turboscan::Proto::AlertResponse` and so should have the same signature for `intput` as the generated DSL, in which we see:

```
class Turboscan::Proto::AlertRequest
  sig do
    params(
      number: T.nilable(Integer),
      ref_names_bytes: T.nilable(T.any(Google::Protobuf::RepeatedField[String], T::Array[String])),
      repository_id: T.nilable(Integer)
    ).void
  end
  def initialize(number: nil, ref_names_bytes: T.unsafe(nil), repository_id: nil); end
```

And for completeness, in the genrated `typed: false` gem RBI we see

```
class Turboscan::Proto::AlertRequest
  include ::Google::Protobuf::MessageExts
  extend ::Google::Protobuf::MessageExts::ClassMethods

  def initialize(*_arg0); end

  ...
```

What I cannot work out how to achieve is how to use the signature for `Turboscan::Proto::AlertRequest` (even if it has to be manual) to keep functions that call `get_alert` up-to-date every time we re-vendor the gem and update the definitions.

What I expected was that there would be a way to ensure that the args provided conform to the correct shape of the generated protobuf ruby, and I have not been able to achieve it. 

I also tried generating a `/rbi` folder in the gem using: `https://github.com/coinbase/protoc-gen-rbi` and that also did not seem to help.

I apologise if this is my own ignorance somehow, but I'm really excited to ensure that the type safety of twirp calls would be something I could ensure.

**Comment by SamMorrowDrums** [comment]
Thanks so much, I'll give it a quick go in my morning.

The signature you suggested is not quite right, in that it actually takes the same shape as the initialise for the AlertRequest, not an initialised AlertRequest itself, and I don't think it's possible to copy the params declaration from one rbi to use on a different function.

I can try the RBI generation again, see if I can succeed.

🙏

**Comment by SamMorrowDrums** [comment]
After some digging, I realised I had not included the RBIs in my gemspec, so they weren't being seen.

So far I'm still struggling to get the gem project itself to run sorbet successfully but that feels like a good start - and I guess until I do that I can't work out if there are other issues.

The project I'm including it in is still doing the following though:
- generating DSL files
- the DSL conflict then results in marking the generated gem RBI file as `typed: false`, which gets further confused when I actually included the RBIs, because further down the file it has `# typed: strict` and things from the included RBIs in the actual gem that seem to get mixed in.

So still failing hard, and not seeing a good way forward 😓 I had to just stop for now.

I'm happy to close the issue seeing as I have a few more things to resolve before I am unblocked, if you prefer?
